app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -398,83 +401,148 @@ private void editMessage(final Message message, final int position) {
-                    //Start a progress dialog indicating editing message
-                    final ProgressDialog progress = new ProgressDialog(getActivity());
-                    progress.setCancelable(false);
-                    progress.setMessage(app.getString(R.string.editing_message));
-                    progress.show();
-                    final String editedMessageContent =
-                            dialogMessageEditText.getText().toString().length() == 0 ? getString(R.string.default_delete_text) : dialogMessageEditText.getText().toString();
-
-                    app.getZulipServices()
-                            .editMessage(String.valueOf(message.getID()), editedMessageContent)
-                            .enqueue(new Callback<EditResponse>() {
-                                @Override
-                                public void onResponse(Call<EditResponse> call, Response<EditResponse> response) {
-                                    if (response.isSuccessful()) {
-                                        message.setContent(editedMessageContent);
-                                        message.setFormattedContent(editedMessageContent);
-                                        adapter.notifyItemChanged(position);
-                                        progress.dismiss();
-                                        Toast.makeText(getActivity(), R.string.message_edited, Toast.LENGTH_SHORT).show();
-                                    } else {
-                                        progress.dismiss();
-                                        Toast.makeText(getActivity(), R.string.message_edit_failed, Toast.LENGTH_SHORT).show();
-                                    }
-                                }
+            final CommonProgressDialog commonProgressDialog = new CommonProgressDialog(getContext());
+            commonProgressDialog.showWithMessage(getString(R.string.fetch_edit_message));
+            app.getZulipServices()
+                    .fetchRawMessage(message.getID())
+                    .enqueue(new DefaultCallback<RawMessageResponse>() {
+                        @Override
+                        public void onSuccess(Call<RawMessageResponse> call, Response<RawMessageResponse> response) {
+                            RawMessageResponse messageResponse = response.body();
+                            commonProgressDialog.dismiss();
+                            showEditMessageDialog(message, messageResponse.getRawContent(), position);
+                        }
+
+                        @Override
+                        public void onError(Call<RawMessageResponse> call, Response<RawMessageResponse> response) {
+                            RawMessageResponse messageResponse = response.body();
+                            Toast.makeText(getActivity(), (TextUtils.isEmpty(messageResponse.getMsg())) ? getString(R.string.message_edit_failed) :
```

Comment:
We might wanna have a check here if messageResponse is null; for eg: in case of an internal server error, the response is null.
I get the following crash we the onError is called
```
E/AndroidRuntime: FATAL EXCEPTION: main
                                                                       Process: com.zulip.android.dev, PID: 30231
                                                                       java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.zulip.android.networking.response.RawMessageResponse.getMsg()' on a null object reference
                                                                           at com.zulip.android.activities.MessageListFragment$6.onError(MessageListFragment.java:528)
                                                                           at com.zulip.android.networking.util.DefaultCallback.onResponse(DefaultCallback.java:20)
                                                                           at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall$1$1.run(ExecutorCallAdapterFactory.java:68)
                                                                           at android.os.Handler.handleCallback(Handler.java:739)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                           at android.os.Looper.loop(Looper.java:148)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:7325)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```